### DISCUSSION TITLE
Can't pull from private Gitlab repository protected by 2-factor authentication

### DISCUSSION THREAD

**Discussion by ghbillings** [nextflow-io/nextflow#2278] [Q&A] [answered]
## Bug report 

I can't pull from private Gitlab repository when my account is protected by 2-factor authentication

### Expected behavior and actual behavior

I am attempting to access a private Gitlab repo for use with Nextflow's pipeline sharing functionality, but am getting the error below; when I disable 2FA, the error goes away.

### Steps to reproduce the problem

I've created a Gitlab access token with api, read_api, and read_repository permissions, and created a `~/.nextflow/scm` file that looks like:
```
providers {

    gitlab {
	user = '<user>'
        password = '<password>'
        token = '<token>'
    }

}
```
Then, I ran `nextflow run https://gitlab.com/<group_name>/nf_test_repo`; the command fails when 2FA is enabled, but the exact same command works when 2FA is disabled.

### Program output 
When run with 2FA enabled, I get the following error
```
nextflow run https://gitlab.com/<group_name>/nf_test_repo
N E X T F L O W  ~  version 21.04.3
Pulling <group_name>/nf_test_repo ...
https://gitlab.com/<group_name>/nf_test_repo.git: authentication not supported
```

### Environment 

* Nextflow version: 21.04.3 
* Java version: openjdk version "1.8.0_152-release"
* Operating system: Amazon Linux 2
* Bash version: 4.2.46(2)-release

### Additional context

I'm surprised I need to enter both password and token: per the [GitLab documentation](https://docs.gitlab.com/ee/user/profile/account/two_factor_authentication.html#personal-access-tokens):
> When 2FA is enabled, you can no longer use your normal account password to authenticate with Git over HTTPS on the command line or when using the GitLab API. You must use a personal access token instead.

**Comment by abhi18av** [comment, marked as answer]
Hi @ghbillings ,

Could you please try again after specifying the value of `token` for `password` field in the `scm` file as well?

**Reply by ghbillings** [reply]
Thank you, that's working for me now!

**Reply by abhi18av** [reply]
You're welcome.
